pycode/memilio-epidata/memilio/epidata/getNPIData.py

```diff
@@ -1196,6 +1216,133 @@ def get_npi_data(fine_resolution=2,
+    return df_count
+
+
+def count_codes_active(df_merged, df_count_active, county):
+    df_local = df_merged[df_merged[dd.EngEng['idCounty']] == county]
+    code_dict = {}
+    for maincode in df_count_active.keys():
+        for column in df_count_active[maincode][1].columns:
+            code_dict[column] = df_local.iloc[np.where(
+                df_local.loc[:, df_local.columns.str.contains(column)].max(axis=1) > 0)[0], 0].to_list()
+
+    # with diag
+    # for maincode in df_count_active.keys():
+    #    column_list = df_count_active[maincode][1].columns
+    #    for column in range(len(column_list)):
+    #        for column_other in range(len(column_list)):
+    #            df_count_active[maincode][1].iloc[column, column_other] += len(set(
+    #                code_dict[column_list[column]]).intersection(set(code_dict[column_list[column_other]])))
+
+    # no diag
+    for maincode in df_count_active.keys():
+        column_list = df_count_active[maincode][1].columns
+        for column in range(len(column_list)):
+            for column_other in range(column):
+                df_count_active[maincode][1].iloc[column, column_other] += len(set(
+                    code_dict[column_list[column]]).intersection(set(code_dict[column_list[column_other]])))
+
+    return df_count_active
+
+
+def count_codes_incid_depend(df_incid_depend, df_count_incid_depend, county):
+    df_local = df_incid_depend[df_incid_depend[dd.EngEng['idCounty']] == county]
+    code_dict = {}
+    for maincode in df_count_incid_depend.keys():
+        for column in df_count_incid_depend[maincode][1].columns:
+            code_dict[column] = df_local.iloc[np.where(
+                df_local.loc[:, df_local.columns.str.contains(column)].max(axis=1) > 0)[0], 0].to_list()
+
+    # with diag
+    # for maincode in df_count_incid_depend.keys():
+    #    column_list = df_count_incid_depend[maincode][1].columns
+    #    for column in range(len(column_list)):
+    #        for column_other in range(len(column_list)):
+    #            df_count_incid_depend[maincode][1].iloc[column, column_other] += len(set(
+    #                code_dict[column_list[column]]).intersection(set(code_dict[column_list[column_other]])))
+
+    # no diag
+    for maincode in df_count_incid_depend.keys():
+        column_list = df_count_incid_depend[maincode][1].columns
+        for column in range(len(column_list)):
+            for column_other in range(column):
+                df_count_incid_depend[maincode][1].iloc[column, column_other] += len(set(
+                    code_dict[column_list[column]]).intersection(set(code_dict[column_list[column_other]])))
+
+    return df_count_incid_depend
+
+
+def save_counter(df_count, filename, directory):
+    # save results
+
+    writer = pd.ExcelWriter(
+        os.path.join(directory, filename + '.xlsx'),
+        engine='xlsxwriter')
```

Comment:
`xlsxwriter` needs to be added to install_requires https://github.com/DLR-SC/memilio/blob/main/pycode/memilio-epidata/setup.py#L73